From 8866ccf9941c4adb7f740f5995cb06a562086542 Mon Sep 17 00:00:00 2001
From: Vaishnav Achath <vaishnav.a@ti.com>
Date: Fri, 23 Sep 2022 19:23:22 +0530
Subject: [PATCH] phy: cdns-dphy: Fix common module reset logic

While inverting the logic for SoC specific common module reset, the
DPHY_LANE_RESET_CMN_EN was performed only when soc_device_match() returns
match, but since the newer SoCs has been removed from the table, this
causes issue with streaming for newer SoCs.This commit fixes the common
module reset properly to not issue software reset only for J721E SR1.0
and issue software RSTB_CMN for all other new platforms.

Fixes: 0f25c34882ff ("phy: cdns-dphy: Update common module reset logic for newer platforms")

Signed-off-by: Vaishnav Achath <vaishnav.a@ti.com>
Reviewed-by: Jai Luthra <j-luthra@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 drivers/phy/cadence/cdns-dphy.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/phy/cadence/cdns-dphy.c b/drivers/phy/cadence/cdns-dphy.c
index 47fd8d206809..68b871111e9d 100644
--- a/drivers/phy/cadence/cdns-dphy.c
+++ b/drivers/phy/cadence/cdns-dphy.c
@@ -553,17 +553,16 @@ static int cdns_dphy_rx_configure(struct cdns_dphy *dphy,
 				  union phy_configure_opts *opts)
 {
 	const struct soc_device_attribute *soc;
-	const struct cdns_dphy_soc_data *soc_data;
+	const struct cdns_dphy_soc_data *soc_data = NULL;
 	unsigned int reg;
 	int band_ctrl, ret;
 
 	soc = soc_device_match(cdns_dphy_socinfo);
-	if (soc && soc->data) {
+	if (soc && soc->data)
 		soc_data = soc->data;
-		if (!soc_data->has_hw_cmn_rstb) {
-			reg = DPHY_LANE_RESET_CMN_EN;
-			writel(reg, dphy->regs + DPHY_LANE);
-		}
+	if (!soc || (soc_data && !soc_data->has_hw_cmn_rstb)) {
+		reg = DPHY_LANE_RESET_CMN_EN;
+		writel(reg, dphy->regs + DPHY_LANE);
 	}
 
 	band_ctrl = cdns_dphy_rx_get_band_ctrl(opts->mipi_dphy.hs_clk_rate);
-- 
2.17.1

